projects/packages/device-detection/src/class-device-detection.php

```diff
@@ -41,6 +55,14 @@ class Device_Detection {
 	 * );
 	 */
 	public static function get_info( $ua = '' ) {
+		// Return memoized result if available.
+		// phpcs:disable WordPress.Security.ValidatedSanitizedInput
+		$memo_key = ! empty( $ua ) ? $ua : ( $_SERVER['HTTP_USER_AGENT'] ?? '' );
+		// phpcs:enable WordPress.Security.ValidatedSanitizedInput
```

Comment:
Is there a specific reason we cannot add some sanitization like `sanitize_text_field()` here? Is it because `get_info` may at times be called super early, before WordPress' sanitization functions are available? Could we use something else instead? If not, maybe it's worth adding a note next to the `phpcs:disable` comment to explain that to our future selves?